### PR TITLE
fix typing for payment and discountCode

### PR DESCRIPTION
**Pull request by mastilver**
Hi

A couple of children resources didn't include the parent id

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -155,12 +155,12 @@ declare class Shopify {
...
     discountCode: {
-        create: (params: any) => Promise<Shopify.IDiscountCode>;
-        delete: (id: number) => Promise<void>;
-        disable: (id: number) => Promise<Shopify.IDiscountCode>;
-        enable: (id: number) => Promise<Shopify.IDiscountCode>;
-        get: (id: number) => Promise<Shopify.IDiscountCode>;
-        list: (params?: any) => Promise<Shopify.IDiscountCode[]>;
+        create: (priceRuleId: number, params: any) => Promise<Shopify.IDiscountCode>;
+        delete: (priceRuleId: number, id: number) => Promise<void>;
+        disable: (priceRuleId: number, id: number) => Promise<Shopify.IDiscountCode>;
```
Comment (lpinca) [inline index.d.ts]: This does not exist.

File: index.d.ts [hunk 2]
```diff
@@ -155,12 +155,12 @@ declare class Shopify {
...
+        enable: (priceRuleId: number, id: number) => Promise<Shopify.IDiscountCode>;
```
Comment (lpinca) [inline index.d.ts]: Ditto.

File: index.d.ts [hunk 3]
```diff
@@ -155,12 +155,12 @@ declare class Shopify {
...
+        get: (priceRuleId: number, id: number) => Promise<Shopify.IDiscountCode>;
+        list: (priceRuleId: number, params?: any) => Promise<Shopify.IDiscountCode[]>;
```
Comment (lpinca) [inline index.d.ts]: `lookup` and `update` are missing.

File: index.d.ts [hunk 4]
```diff
@@ -271,10 +271,10 @@ declare class Shopify {
         update: (id: number, params: any) => Promise<Shopify.IPage>;
     };
     payment: {
-        count: (checkoutToken: any) => Promise<number>;
-        create: (checkoutToken: any, params: any) => Promise<any>;
-        get: (checkoutToken: any, id: number) => Promise<any>;
-        list: (checkoutToken?: any) => Promise<any>;
+        count: (checkoutId:number, checkoutToken: any) => Promise<number>;
```
Comment (lpinca) [inline index.d.ts]: No changes are needed here as per https://help.shopify.com/en/api/reference/sales_channels/payment

File: index.d.ts [hunk 5]
```diff
@@ -276,7 +275,7 @@ declare class Shopify {
     payment: {
         count: (checkoutToken: any) => Promise<number>;
         create: (checkoutToken: any, params: any) => Promise<any>;
-        get: (checkoutToken: any, id: number) => Promise<any>;
+        get: (checkoutId:number, checkoutToken: any, id: number) => Promise<any>;
```
Comment (lpinca) [inline index.d.ts]: This doesn't look correct, I think it shouldn't be changed, see https://help.shopify.com/en/api/reference/sales_channels/payment#show

File: index.d.ts [hunk 6]
```diff
@@ -1555,7 +1554,7 @@ declare namespace Shopify {
         name: string;
         value: string
     }
-                              
+
```
Comment (lpinca) [inline index.d.ts]: ❤️

File: index.d.ts (hunk 5 above)
Comment (mastilver) [reply index.d.ts]: Right, somehow missed the id...

BTW, `checkoutToken` should be a `string`, right or is there a reason it's `any` right now?

Comment (lpinca) [reply index.d.ts]: Yes you are right, I think there is no particular reason, thanks for fixing it.

**Comment by lpinca** [comment]
Thank you.

**Pull request merged**